k8s/k8s.go

```diff
@@ -0,0 +1,851 @@
+	ostclientset "github.com/libopenstorage/operator/pkg/client/clientset/versioned"
+	storkclientset "github.com/libopenstorage/stork/pkg/client/clientset/versioned"
+	ocp_clientset "github.com/openshift/client-go/apps/clientset/versioned"
+	ocp_security_clientset "github.com/openshift/client-go/security/clientset/versioned"
+	"github.com/portworx/sched-ops/k8s/admissionregistration"
+	"github.com/portworx/sched-ops/k8s/apiextensions"
+	"github.com/portworx/sched-ops/k8s/apps"
+	"github.com/portworx/sched-ops/k8s/autopilot"
+	"github.com/portworx/sched-ops/k8s/batch"
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/portworx/sched-ops/k8s/discovery"
+	"github.com/portworx/sched-ops/k8s/dynamic"
+	"github.com/portworx/sched-ops/k8s/externalstorage"
+	"github.com/portworx/sched-ops/k8s/openshift"
+	"github.com/portworx/sched-ops/k8s/operator"
+	"github.com/portworx/sched-ops/k8s/prometheus"
+	"github.com/portworx/sched-ops/k8s/rbac"
+	"github.com/portworx/sched-ops/k8s/storage"
+	"github.com/portworx/sched-ops/k8s/stork"
+	"github.com/portworx/sched-ops/k8s/talisman"
+	talismanclientset "github.com/portworx/talisman/pkg/client/clientset/versioned"
+	hook "k8s.io/api/admissionregistration/v1beta1"
+	appsv1 "k8s.io/api/apps/v1"
+	batch_v1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	rbac_v1 "k8s.io/api/rbac/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	storagev1beta1 "k8s.io/api/storage/v1beta1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/version"
+	dynamicclient "k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// Ops is an interface to perform any kubernetes related operations
+type Ops interface {
+	admissionregistration.Interface
+	apiextensions.Interface
+	apps.Interface
+	autopilot.Interface
+	batch.Interface
+	core.Interface
+	discovery.Interface
+	externalstorage.Interface
+	openshift.Interface
+	operator.Interface
+	prometheus.Interface
+	rbac.Interface
+	storage.Interface
+	stork.Interface
+	talisman.Interface
+	ClientSetter
+}
+
+var (
+	instance Ops
+	once     sync.Once
+)
+
+type k8sOps struct {
+	client             kubernetes.Interface
+	snapClient         rest.Interface
+	storkClient        storkclientset.Interface
+	ostClient          ostclientset.Interface
+	talismanClient     talismanclientset.Interface
+	autopilotClient    autopilotclientset.Interface
+	apiExtensionClient apiextensionsclient.Interface
+	dynamicInterface   dynamicclient.Interface
+	ocpClient          ocp_clientset.Interface
+	ocpSecurityClient  ocp_security_clientset.Interface
+	prometheusClient   prometheusclient.Interface
+}
+
+// Instance returns a singleton instance of k8sOps type
+func Instance() Ops {
+	once.Do(func() {
+		instance = &k8sOps{}
```

Comment:
From the above context, here we should just initialize all individual clients and not set the singleton instances of it.
```
instance = &k8sOps{
    CoreClient: &core.CoreClient{},
    StorkClient: &stork.StorkClient{},
    ...
}
```
I feel users should just avoid using the new and old way of using the k8s ops. It should be either the old way or the new split clients.

Reply:
Updated.